wire-compiler/src/main/java/com/squareup/wire/KotlinFileWriter.kt

```diff
@@ -26,29 +28,22 @@ import java.util.concurrent.ConcurrentLinkedQueue
 internal class KotlinFileWriter(
   private val destination: String,
   private val kotlinGenerator: KotlinGenerator,
-  private val queue: ConcurrentLinkedQueue<Type>,
+  private val queue: ConcurrentLinkedQueue<Any>,
```

Comment:
Can we make Service extend Type? I wonder what other assumptions that’d break

Reply:
I'll try to move `Service` to `ServiceType : Type` in followup today

Reply:
I don't like that at all. It breaks the soundness of the model since things like a Message can declare a field with a Service as the type.

The need to process things that don't have an is-a relationship with each other means you need to create one in the domain where that relationship exists. That is in this code, not in the model. Define a sealed hierarchy for things that need processed with a method:
```kotlin
sealed class PendingFile {
  abstract fun writeTo(directory: Path)

  data class TypeFile(val type: Type) : PendingFile() {
    override fun writeTo(directory: Path) = TODO()
  }
  data class ServiceFile(val service: Service) : PendingFile() {
    override fun writeTo(directory: Path) = TODO()
  }
}
```

Reply:
We’ve got two concepts that are mutually overlapping:

 * Named things that can be byte-encoded: messages, enums, map, and scalars
 * Named things that contain declarations: messages, enums, services, enclosing types

What we currently call `Type` is neither the first nor the second.